index.js

```diff
@@ -1,19 +1,52 @@
 const fs = require('fs')
+const moment = require('moment')
 
 const logSession = new Date().toISOString()
 const logStream = fs.createWriteStream('/tmp/express-requests-' + logSession + '.log', {flags:'a'})
 
-let _requests = 0
+const interval = 3000 // milliseconds to check number of requests
+const limit = 3 // limit for requests within interval
+let _rateLimitingAddress2Requests = {}
 
-module.exports = {
-  countRequests: function(req, res, next) {
-    _requests++
-    const now = new Date()
-    logStream.write('Connection requested at: ' + now + ', connection attempt: ' + _requests + '\n')
-    next()
-  },
-  getRequests: function() {
-    const requests = _requests
-    return requests
+const formatMoment = (moment) => {
+  return moment.format()
+}
+
+const addAddress2Requests = (address, requests, startRequestAt) => {
+  _rateLimitingAddress2Requests[address] = { requests: requests, startRequestAt: startRequestAt }
```

Comment:
Possible to simplify to ` _rateLimitingAddress2Requests[address] = { requests, startRequestAt }
`